File: 5Days-SoftwareArchitecture/typesript/Day3/white-box-testing/basis_path_testing.js
var BankService = /** @class */ (function () {
    function BankService() {
    }
    BankService.prototype.withdraw = function (balance, amount, otpVerified) {
        if (balance <= 0) {
            console.log("No balance");
        }
        else if (amount > balance) {
            console.log("Insufficient balance");
        }
        else if (!otpVerified) {
            console.log("OTP not verified");
        }
        else {
            console.log("Withdraw successful");
        }
    };
    return BankService;
}());
// Create an instance
var bank = new BankService();
// Test cases
console.log("=== Test Case 1: No balance ===");
bank.withdraw(0, 100, true);
console.log("\n=== Test Case 2: Insufficient balance ===");
bank.withdraw(500, 1000, true);
console.log("\n=== Test Case 3: OTP not verified ===");
bank.withdraw(1000, 500, false);
console.log("\n=== Test Case 4: Successful withdrawal ===");
bank.withdraw(1000, 500, true);

File: 5Days-SoftwareArchitecture/Java/Day3/White-box-testing/basis_path_testing.java
class BankService {

    public void withdraw(double balance, double amount, boolean otpVerified) {
        if (balance <= 0) {
            System.out.println("No balance");
        } else if (amount > balance) {
            System.out.println("Insufficient balance");
        } else if (!otpVerified) {
            System.out.println("OTP not verified");
        } else {
            System.out.println("Withdraw successful");
        }
    }
}

public class basis_path_testing {
    public static void main(String[] args) {
        BankService bank = new BankService();

        System.out.println("=== Test Case 1: No balance ===");
        bank.withdraw(0, 100, true);

        System.out.println("\n=== Test Case 2: Insufficient balance ===");
        bank.withdraw(500, 1000, true);

        System.out.println("\n=== Test Case 3: OTP not verified ===");
        bank.withdraw(1000, 500, false);

        System.out.println("\n=== Test Case 4: Successful withdrawal ===");
        bank.withdraw(1000, 500, true);
    }
}
